### PR TITLE
feat: Add field-level immutability to the metadata.

### PR DESCRIPTION
**Pull request by stephenh**
In theory this will let us short-circuit field-level validation rules,
should those exist in the future.

### CODE REVIEW

File: packages/integration-tests/src/entities/Author.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class Author extends AuthorCodegen {
   public afterCommitRan = false;
   public afterCommitIdIsSet = false;
   public afterCommitIsNewEntity = false;
-  public ageForBeforeFlush?: number;
+  public setGradatedInFlush?: boolean;
```
Comment (chr1sjf0x) [inline packages/integration-tests/src/entities/Author.ts:48]: quibble: 
```suggestion
  public setGraduatedInFlush?: boolean;
```

File: packages/integration-tests/src/entities/Author.ts [hunk 2]
```diff
@@ -122,15 +122,15 @@ config.addRule("books", async (a) => {
 });
 
 // Example of cannotBeUpdated
-config.addRule(cannotBeUpdated("age", (a) => !!a.age && a.age < 100));
+config.addRule(cannotBeUpdated("age"));
```
Comment (chr1sjf0x) [inline packages/integration-tests/src/entities/Author.ts:125]: thought: I was half expecting this to look something like 
```suggestion
config.cannotBeChanged("age");
```
But after further digging I see how you are going about it...

And you plan to do use this marker in the future for possible short circuiting 🤔  Interesting.

**Review by chr1sjf0x — APPROVED** [review]

File: packages/integration-tests/src/entities/Author.ts (hunk 1 above)
Comment (stephenh) [reply packages/integration-tests/src/entities/Author.ts]: Fixed, thanks!

**Pull request merged**